src/EnergyPlus/ScheduleManager.cc

```diff
@@ -1641,354 +1636,229 @@ namespace ScheduleManager {
+                vectorOfCSVColNumbers
+                    .end()) { // TODO : what if there are multiple last elements? - remove last element and check if duplicate is still the same
+                ShowFatalError(state,
+                               format("Multiple schedules pointing to the same column number : {} in file {}. Please modify the IDF file and make "
+                                      "sure multiple schedules do not point to the same column in the file.",
+                                      *duplicate,
+                                      fileNameItem));
+            }
+
+            std::ifstream file(fileNameItem);
+            CSVRow row;
+            // TODO : Fix delimiter issue: check if there are other delimiters specified, if yes, throw error
+            row.delimiter = state.dataScheduleMgr->columnarData[colNumToColDataIndex[vectorOfCSVColNumbers[0]]]
+                                .delimiter; // all schedules in one file will have the same delimiter
+
+            while (file >> row)
+            // find which columns of this csv map to which schedule - not needed
+            //  How to do this? Make a mapOfMaps which has for each fileName, a map of scheduleNames, ColumnNumbers
+
+            {
+                for (int colNum : vectorOfCSVColNumbers) {
+
+                    // skip if row needs to be skipped
+                    if (state.dataScheduleMgr->columnarData[colNumToColDataIndex[colNum]].rowsToSkip) {
+                        --state.dataScheduleMgr->columnarData[colNumToColDataIndex[colNum]].rowsToSkip;
+                        continue;
+                    } // TODO : Add check if the number of rows skipped were correct (What if the user input was wrong?)
+
+                    columnValue = UtilityRoutines::ProcessNumber(row[colNum], errFlag);
+                    if (errFlag) {
+                        ++state.dataScheduleMgr->columnarData[colNumToColDataIndex[colNum]].numerrors;
+                        columnValue = 0.0;
+                    }
 
-            // Depending on value of "Interpolate" field, the value for each time step in each hour gets processed:
-            FileIntervalInterpolated = false;
-            if (lAlphaBlanks(5)) Alphas(5) = "NO";
-            if (Alphas(5) != "NO" && Alphas(5) != "YES") {
-                ShowSevereError(state,
-                                RoutineName + CurrentModuleObject + "=\"" + Alphas(1) + "Invalid value for \"" + cAlphaFields(5) + "\" field=\"" +
-                                    Alphas(5) + "\"");
-                ErrorsFound = true;
-            } else if (Alphas(5) != "YES") { // No validation done on the value of the interpolation field
-                FileIntervalInterpolated = false;
-            } else {
-                FileIntervalInterpolated = true;
-            }
+                    state.dataScheduleMgr->columnarData[colNumToColDataIndex[colNum]].vals.emplace_back(columnValue); // row[colNum]
```

Comment:
Go through every row of the file, 
For every column number that we know are important,
  if it needs to be skipped, skip it.
Or else, process the token or cell to a Real64 value and 
put it in the appropriate vals vector.